src/Ownership/Readmodels/OwnershipSearchProjector.php

```diff
@@ -76,4 +95,74 @@ public function applyOwnershipDeleted(OwnershipDeleted $ownershipDeleted): void
             OwnershipState::deleted()
         );
     }
+
+    protected function applyConstraintAdded(ConstraintAdded $constraintEvent): void
+    {
+        $this->processConstraint($constraintEvent->getUuid(), $constraintEvent->getQuery());
+    }
+
+    protected function applyConstraintUpdated(ConstraintUpdated $constraintEvent): void
+    {
+        $roleId = $constraintEvent->getUuid();
+
+        $this->connection->beginTransaction();
+
+        // First clear all existing owners for this role, because if the constraint is changed some users might no longer be valid.
+        $this->ownershipSearchRepository->deleteByRole($roleId);
```

Comment:
I wonder if this can cause some side effects if an existing ownership gets deleted? This would happen if a role attached to a "normal" ownership gets updated.

Reply:
if a normal role is updated, there will be no ownership attached to it.
I don't immediately see the problem, in which direction are you thinking?

Reply:
What is a normal role? Because an ownership also has a role, so how do you see the difference?